src/VirtualClient/VirtualClient.Actions/HadoopTerasort/HadoopTerasortExecutor.cs

```diff
@@ -0,0 +1,389 @@
+                await this.ExecuteCommandAsync("bash", $"-c \"bin/hdfs dfs -mkdir /user\"", this.PackageDirectory, telemetryContext, cancellationToken).ConfigureAwait(false);
+                await this.ExecuteCommandAsync("bash", $"-c \"bin/hdfs dfs -mkdir /user/{Environment.UserName}\"", this.PackageDirectory, telemetryContext, cancellationToken).ConfigureAwait(false);
+
+                this.state.HadoopExecutorStateInitialized = true;
+            }
+
+            if (!this.state.HadoopExecutorServicesStarted)
+            {
+                await this.ExecuteCommandAsync("bash", $"-c sbin/start-dfs.sh", this.PackageDirectory, telemetryContext, cancellationToken).ConfigureAwait(false);
+                await this.ExecuteCommandAsync("bash", $"-c sbin/start-yarn.sh", this.PackageDirectory, telemetryContext, cancellationToken).ConfigureAwait(false);
+                
+                this.state.HadoopExecutorServicesStarted = true;
+
+                // Delay during the safe mode of the name node to verify the status of replicated nodes.
+                // For operations that includes executing tasks rapidly, encompassing swift initiation and termination, is crucial.
+                // A delay helps avoid potential failures in the process of loading data into the service.
+                // await Task.Delay(30000); 
+            }
+
+            await this.stateManager.SaveStateAsync<HadoopExecutorState>($"{nameof(HadoopExecutorState)}", this.state, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Returns true/false whether the component is supported on the current
+        /// OS platform and CPU architecture.
+        /// </summary>
+        protected override bool IsSupported()
+        {
+            bool isSupported = base.IsSupported()
+                && (this.Platform == PlatformID.Unix)
+                && (this.CpuArchitecture == Architecture.X64);
+
+            if (!isSupported)
+            {
+                this.Logger.LogNotSupported("Hadoop Terasort", this.Platform, this.CpuArchitecture, EventContext.Persisted());
+            }
+
+            return isSupported;
+        }
+
+        /// <summary>
+        /// Logs the Hadoop Terasort workload metrics.
+        /// </summary>
+        private void CaptureMetrics(IProcessProxy process, string toolName, EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            if (!cancellationToken.IsCancellationRequested)
+            {
+                this.MetadataContract.AddForScenario(
+                    toolName,
+                    process.FullCommand(),
+                    toolVersion: null);
+
+                this.MetadataContract.Apply(telemetryContext);
+
+                // Hadoop workload produces metrics in standard 
+                HadoopMetricsParser parser = new HadoopMetricsParser(process.StandardError.ToString());
```

Comment:
What's in the standard out? are they more useful than standard error? It's really bad for them to put metrics in standarderror

Reply:
The Hadoop service shares the execution results via standard error. Thus, I had to use it.